hibernate-core/src/main/java/org/hibernate/id/CompositeNestedGeneratedValueGenerator.java

```diff
@@ -129,7 +128,17 @@ public Object generate(SharedSessionContractImplementor session, Object object)
 				null :
 				new ArrayList<>( generationPlans.size() );
 		for ( GenerationPlan generationPlan : generationPlans ) {
-			final Object generated = generationPlan.execute( session, object );
+			final BeforeExecutionGenerator generator = generationPlan.getGenerator();
+			final Object generated;
+			if ( generator.generatedBeforeExecution( object, session ) ) {
+				final Object currentValue = generator.allowAssignedIdentifiers() ?
+						compositeType.getPropertyValue( context, generationPlan.getPropertyIndex(), session ) :
+						null;
```

Comment:
```suggestion
				final Object currentValue = generator.allowAssignedIdentifiers()
						? compositeType.getPropertyValue( context, generationPlan.getPropertyIndex(), session )
						: null;
```

Sorry for the nitpick!